src/Lhm/Chunker.php

```diff
@@ -122,7 +152,28 @@ protected function execute()
         $initialStart = microtime(true);
         $lastUpdate = 0;
         $totalRows = 0;
+        $lastSlaveLagCheck = 0;
+
         while ($this->currentOffset <= $this->limit || $this->currentOffset === 0) {
+            if ($lastSlaveLagCheck < time() - self::SLAVE_LAG_CHECK_FREQUENCY
+                && !empty($this->slaveAdapters)
+            ) {
+                foreach ($this->slaveAdapters as $slaveAdapter) {
+                    $result = $slaveAdapter->query('SHOW SLAVE STATUS')->fetchAll();
+                    foreach ($result as $row) {
+                        $lag = $result['Seconds_Behind_Master'];
+                        if ($lag > self::ALLOWED_SLAVE_LAG) {
+                            $this->logger->warning("Slave lag at $lag. Over allowed max, slowing down");
+                            sleep(10);
+                            $stride /= 2;
```

Comment:
So I was wrong and said to change the stride, that's not what LHM/pt-osc are doing.

They ramp up or lower the sleep time between queries (e.g. sleep a minimum of 1ms, a maximum of 10s) depending on if the slaves are behind.

So something like:
```
$sleepTime = 1000;
if (overSlaveLag) {
    $sleepTime = min($sleepTime * 2, 1000000);
} else {
    $sleepTime = max($sleepTime / 2, 1000);
}
usleep($sleepTime);
```